src/CacheTower.Providers.Redis/RedisCacheLayer.cs

```diff
@@ -32,11 +33,18 @@ public class RedisCacheLayer : ICacheLayer
 		/// The database index to use for Redis.
 		/// If not specified, uses the default database as configured on the <paramref name="connection"/>.
 		/// </param>
-		public RedisCacheLayer(IConnectionMultiplexer connection, int databaseIndex = -1)
+		/// <param name="serializer">
+		/// Allows you to specify which encoding should be used by providing your own serializer
+		/// If one is not provided, ProtobufCacheSerializer will be used
+		/// </param>
+		public RedisCacheLayer(IConnectionMultiplexer connection, int databaseIndex = -1, ICacheSerializer? serializer = null)
```

Comment:
So I was thinking rather than modifying the existing constructor, we create a new one.

```csharp
public RedisCacheLayer(IConnectionMultiplexer connection, int databaseIndex = -1) : this(connection, new ProtobufCacheSerializer(), databaseIndex)

...

public RedisCacheLayer(IConnectionMultiplexer connection, ICacheSerializer serializer, int databaseIndex = -1)
```

Then in the future, we obsolete the original constructor.